src/solvers/dgsem_tree/indicators.jl

```diff
@@ -212,6 +212,85 @@ end
 
 
 
+"""
+    IndicatorIDP(equations::AbstractEquations, basis;
+                 IDPPositivity=false,
+                 variables_cons=(first,),
+                 positCorrFactor=0.1)
+
+Blending indicator used for subcell shock-capturing [`VolumeIntegralShockCapturingSubcell`](@ref) including:
+- positivity limiting for conservative variables.
+
+## References
+
+- Rueda-Ramírez, Pazner, Gassner (2022)
+  "Subcell Limiting Strategies for Discontinuous Galerkin Spectral Element Methods"
+- Pazner (2020)
+  "Sparse invariant domain preserving discontinuous Galerkin methods with subcell convex limiting"
+  [arXiv:2004.08503](https://doi.org/10.1016/j.cma.2021.113876)
+
+!!! warning "Experimental implementation"
+    This is an experimental feature and may change in future releases.
+"""
+struct IndicatorIDP{RealT<:Real, LimitingVariablesCons, Cache} <: AbstractIndicator
+  IDPPositivity::Bool
+  variables_cons::LimitingVariablesCons # Positivity of conservative variables
+  cache::Cache
+  positCorrFactor::RealT                # Correction factor for IDPPositivity
+end
+
+# this method is used when the indicator is constructed as for shock-capturing volume integrals
+function IndicatorIDP(equations::AbstractEquations, basis;
+                      IDPPositivity=false,
+                      variables_cons=(first,),
```

Comment:
Does it make sense do have a default here? I'd say, if someone uses this indicator, they always should be explicit about what should be limited (and not accidentally limit the first variable)

Reply:
Since you explicitly activate the positivity limiter by `IDPpositivity=true` to get any limiting at all, I thought it might be helpful to add a default variable. But I can delete it as well.

Reply:
No, this is really up to you to decide: If you think there are enough cases where `first` is a good choice, i.e., if there exist enough setups where people will benefit from this default, you should definitely leave it. I am not an expert here, it just felt odd to me because the "first" variable of a system has no special meaning usually